src/common/http.cc

```diff
@@ -922,6 +923,33 @@ namespace Pistache::Http
         }
 #endif
 
+#ifdef PISTACHE_USE_CONTENT_ENCODING_ZSTD
+
+        case Http::Header::Encoding::Zstd: {
+            // Get max compressed size
+            size_t estimated_size = ZSTD_compressBound(size);
+            // Allocate a smart buffer to contain compressed data...
+            std::unique_ptr compressedData = std::make_unique<std::byte[]>(estimated_size);
+
+            // Compress data using compresion_level = 5: https://facebook.github.io/zstd/zstd_manual.html#Chapter5
```

Comment:
We currently don't have a public API for the user to specify the compression level (each compressor can use a different range). Until we figure out an elegant way to do that, it might be best to set the compression level consistent with whatever we're using on the others (I think max).

Reply:
I believe max is _really_ max on zstd. We could also set compression level dynamically, or partially dynamically, taking into account available CPU bandwidth, available network bandwidth, number of cores (zstd is good with multithreading) etc.

But maybe a middling value is wisest until we do a well-thought-through API/algorithm for compression level?

Reply:
@dgreatwood @kiplingw I agree. Using a middle value makes the most sense to me. At least until we find a better solution as both of you guys mentioned.